### PR TITLE
py-pythran: update to 0.15.0

### PR DESCRIPTION
**Pull request by erikbs**
#### Description

Update to 0.15.0. Remove patch that has been merged upstream.

This also unbreaks the py312 subport (it currently compiles, but does not actually support Python 3.12). Tested by building SciPy with it.

###### Type(s)

- [ ] bugfix
- [x] enhancement
- [ ] security fix

###### Tested on
macOS 10.9.5 13F1911 x86_64
Xcode 6.2 6C131e

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [x] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

Port has no tests enabled.

### CODE REVIEW

File: python/py-pythran/Portfile [hunk 1]
```diff
@@ -4,7 +4,7 @@ PortSystem          1.0
 PortGroup           python 1.0
 
 name                py-pythran
-version             0.14.0
+version             0.15.0
 revision            1
```
Comment (reneeotten) [inline python/py-pythran/Portfile:8]: the "revision" should be reset to "0" when updating the version

**Review by reneeotten — CHANGES_REQUESTED** [review]
Also, the explicit "depends_build" should be removed as it is added by default due to PEP517.

**Comment by erikbs** [comment]
Will unblock #21953.

File: python/py-pythran/Portfile (hunk 1 above)
Comment (erikbs) [reply python/py-pythran/Portfile]: Thanks, I forgot that. I pushed a new commit with both suggestions.

**Pull request merged**